### PR TITLE
search jobs: add boilerplate to download results

### PR DESCRIPTION
**Pull request by stefanhengl**
This registers the handler to download the results of a search job in csv format. For now we only return dummy data.

Test plan:

I manually tested the following workflow
- Create search job in API Console
```graphql
mutation {
  createSearchJob(query: "1@rev1 1@rev2 2@rev3 3@rev5") {
    id
  }
}
```
- Use the id to request the URL
```graphql
query {
  node(id:"U2VhcmNoSm9iOjMy") {
    ... on SearchJob {
      URL
    }
  }
}
```
- Download the file
```
curl -JO -H 'Authorization: token <TOKEN>' <URL>
```
  - check the downloaded csv

### CODE REVIEW

File: enterprise/cmd/frontend/internal/search/httpapi/export.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		_, err = store.GetExhaustiveSearchJob(r.Context(), jobID)
+		if err != nil {
+			if errors.Is(err, auth.ErrMustBeSiteAdminOrSameUser) {
+				http.Error(w, err.Error(), http.StatusForbidden)
+				return
+			} else {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+		}
```
Comment (keegancsmith) [inline enterprise/cmd/frontend/internal/search/httpapi/export.go:37]: Can you move this sort of business logic into the service package and try make the http handler as lightweight / http specific as possible?

File: enterprise/cmd/frontend/internal/search/httpapi/export.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/sourcegraph/sourcegraph/internal/auth"
+	"github.com/sourcegraph/sourcegraph/internal/search/exhaustive/store"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+func ServeSearchJobDownload(store *store.Store) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		nodeID := mux.Vars(r)["id"]
+
+		jobID, err := resolvers.UnmarshalSearchJobID(graphql.ID(nodeID))
```
Comment (keegancsmith) [inline enterprise/cmd/frontend/internal/search/httpapi/export.go:21]: Kinda weird to use a graphql id here?

**Review by keegancsmith — APPROVED** [review]

File: enterprise/cmd/frontend/internal/search/httpapi/export.go (hunk 2 above)
Comment (stefanhengl) [reply enterprise/cmd/frontend/internal/search/httpapi/export.go:21]: I agree, but the client doesn't has access to the job id, unless we decode the GraphQL ID in the URL resolver and return an URL with an int64 job-id. Do you prefer?

Comment (keegancsmith) [reply enterprise/cmd/frontend/internal/search/httpapi/export.go:21]: Yeah the latter sounds good to me. Doesn't the resolver have direct access to the job struct?

Comment (keegancsmith) [reply enterprise/cmd/frontend/internal/search/httpapi/export.go:21]: wait, just looked at the code for the URL resolver. Isn't it already just using the normal job id integer?

Comment (stefanhengl) [reply enterprise/cmd/frontend/internal/search/httpapi/export.go]: yup, that's a bug. Thanks for catching. Now we accept int64 job ids in the handler.

File: enterprise/cmd/frontend/internal/search/httpapi/export.go (hunk 1 above)
Comment (stefanhengl) [reply enterprise/cmd/frontend/internal/search/httpapi/export.go:34]: discussed synchronously. I will merge as-is and follow up in the PR that connects the handler to the blobstore.

File: enterprise/cmd/frontend/internal/search/resolvers/search_job.go [hunk 3]
```diff
@@ -16,7 +19,7 @@ import (
 
 const searchJobIDKind = "SearchJob"
 
-func unmarshalSearchJobID(id graphql.ID) (int64, error) {
+func UnmarshalSearchJobID(id graphql.ID) (int64, error) {
```
Comment (keegancsmith) [inline enterprise/cmd/frontend/internal/search/resolvers/search_job.go:22]: minor: this doesn't need to be exported anymore.

**Pull request merged**